### PR TITLE
Add correct return type to DependentFixtureInterface::getDependencies

### PR DESCRIPTION
**Pull request by ruudk**
This makes it easier for PHPStan to understand what's going on.

### CODE REVIEW

**Comment by alcaeus** [comment]
How does PhpStorm handle the class-string type? I’m worried that it may not understand what’s going on...

**Comment by ruudk** [comment]
I checked. It just sees it as array of mixed, and that was already the case :)

Screenshot 2019-12-28 at 11 48 42

**Comment by ruudk** [comment]
I'm sure PHPStorm will soon support all the new PHPStan/Psalm notations.

**Comment by alcaeus** [comment]
Great. I’ll take care of this after the holidays, as I need to check the branches and do some more work here for the persistence deprecations.

I believe this should be fixed in a patch release, would you agree?

**Comment by ruudk** [comment]
Thanks! I agree.

**Comment by greg0ire** [comment]
@ruudk this means you shouldn't target master

**Comment by ruudk** [comment]
@greg0ire Sorry, didn't get the hint. Changed the base branch now.

**Comment by greg0ire** [comment]
Triggering a new build, since #333 fixed it

File: lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php [hunk 1]
```diff
@@ -13,7 +13,7 @@ interface DependentFixtureInterface
      * This method must return an array of fixtures classes
      * on which the implementing class depends on
      *
-     * @return array
+     * @return class-string[]
```
Comment (garak) [inline lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php]: I think this should be `@return FixtureInterface[]`

Comment (ruudk) [reply lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php]: I thought so too, but it turns out it expects an array of FQCN class names to do the sorting. Not the actual instances.

Comment (ruudk) [reply lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php]: https://github.com/doctrine/data-fixtures/blob/608a35a3b5bcc4214d116603095f8b0c51091592/lib/Doctrine/Common/DataFixtures/Loader.php#L170-L176

Comment (garak) [reply lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php]: So, it should be `@return \Doctrine\Common\DataFixtures\DependentFixtureInterface[]|\Doctrine\Common\DataFixtures\FixtureInterface[]`

Comment (garak) [reply lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php]: That's a bit of mess, I really don't get why other interfaces are not extending `FixtureInterface`... :disappointed:

Comment (ruudk) [reply lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php]: It should not return instances of that type, it should return FQCN (strings) of fixtures to load first.

Comment (garak) [reply lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php]: ah, you're right. My bad

**Comment by ruudk** [comment]
@greg0ire Can we merge it? :)

**Comment by greg0ire** [comment]
I'm not sure: 1.3 is still maintained, isn't it affected by this too?

**Comment by ruudk** [comment]
@greg0ire Changed base branch and rebased it :shipit:

**Comment by greg0ire** [comment]
Aaaaand the test fail… please cherry-pick b7532e4, that should fix it.

**Comment by ruudk** [comment]
Sorry, but that's not part of this PR. Clearly 1.3.x- branch is broken. I tried to cherry pick it but that forces me to fix a merge conflict. This such a small change, that I already spend too much time one.

**Comment by greg0ire** [comment]
Ok don't worry, I'm going to take care of fixing the branch, just leave this open.

**Comment by greg0ire** [comment]
Ok so apparently it is no longer maintained, so targeting 1.4.x was the right thing to do, sorry.

**Comment by greg0ire** [comment]
Thanks @ruudk !

**Comment by vitek-rostislav** [comment]
Hi, this made my phpstan in version `0.11.6` fail on
```
invalid type Doctrine\Common\DataFixtures\class
```
:disappointed: 

Luckily, upgrading to `^0.12` version helped.

**Pull request merged**